pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -30,88 +30,99 @@ assertExecutable() {
 makeWrapper() {
     local original="$1"
     local wrapper="$2"
-    local params varName value command separator n fileNames
-    local argv0 flagsBefore flags
+    local -a flagsBefore=()
 
     assertExecutable "$original"
 
     mkdir -p "$(dirname "$wrapper")"
 
-    echo "#! $SHELL -e" > "$wrapper"
+    echo "#! $SHELL -eu" > "$wrapper"
+    echo "declare -a extraFlagsArray=()" >> "$wrapper"
 
-    params=("$@")
-    for ((n = 2; n < ${#params[*]}; n += 1)); do
-        p="${params[$n]}"
-
-        if [[ "$p" == "--set" ]]; then
-            varName="${params[$((n + 1))]}"
-            value="${params[$((n + 2))]}"
-            n=$((n + 2))
+    while (( $# )); do
```

Comment:
I don't know if this construct is really clear. Is `(( 0 ))` returning a different status than other numbers? `[[ $# > 0 ]]` might be clearer.

Reply:
I suppose that is clearer; I've been using `(( $# ))` in a number such places for arg processing, so it's become idiomatic to me, but I am not the target audience of my own style of course.

Reply:
Yeah it depends which subset of bash you are using. I have been doing bash for quite a bit and am not familiar with that construct but it doesn't mean I am the norm either ^_^